src/proxy/util.rs

```diff
@@ -22,3 +25,89 @@ pub fn is_runtime_shutdown(e: &Error) -> bool {
     }
     false
 }
+
+#[derive(Default)]
+pub struct Baggage {
+    pub cluster_id: Option<String>,
+    pub namespace: Option<String>,
+    pub workload_name: Option<String>,
+    pub service_name: Option<String>,
+    pub revision: Option<String>,
+}
+
+pub fn parse_baggage_header(h: Option<&HeaderValue>) -> Baggage {
```

Comment:
can we add a fuzzer? see fuzz/ folder and https://github.com/rust-fuzz/cargo-fuzz, its pretty straightforward

Reply:
Just to make sure I follow your line of thoughts about the need for fuzz - Is it because it exposes the ztunnel to attacks over the hbone port?

Reply:
Any custom parsing of untrusted data should have fuzzing, this is one of the top sources of vulnerabilities and its ~trivial to prevent with a 5 line fuzz test